rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_participant_info.hpp

```diff
@@ -235,16 +273,115 @@ class ParticipantListener : public eprosima::fastdds::dds::DomainParticipantList
+        } else {
+          auto hostname = static_peers_string.substr(start, end - start);
+          result.insert(hostname);
+          auto peer_aliases = rmw_fastrtps_shared_cpp::utils::get_peer_aliases(hostname);
+          for (const auto &a : peer_aliases) {
+            result.insert(a);
+          }
+          RCUTILS_LOG_DEBUG_NAMED("rmw_fastrtps_shared_cpp", "Got static peer: %s", hostname.c_str());
+        }
+        start = end + delimiter.length();
+      }
+      if (start < static_peers_string.length() - 1) {
+        auto peer = static_peers_string.substr(start);
+        result.insert(peer);
+        RCUTILS_LOG_DEBUG_NAMED("rmw_fastrtps_shared_cpp", "Got last static peer: %s", peer.c_str());
+      }
+    }
+
+    return result;
+  }
+
+  bool should_ignore_host(const std::string &hostname,
+                          const std::unordered_set<std::string> &other_static_peers) {
+    bool should_ignore = false;
+
+    if (RMW_AUTOMATIC_DISCOVERY_RANGE_OFF ==
+          discovery_options_.automatic_discovery_range) {
+      return true;
+    }
+    if (hostname != my_hostname_) {
+      if (RMW_AUTOMATIC_DISCOVERY_RANGE_LOCALHOST ==
+              discovery_options_.automatic_discovery_range ||
+          RMW_AUTOMATIC_DISCOVERY_RANGE_DEFAULT ==
+              discovery_options_.automatic_discovery_range) {
+
+        if (!is_static_peer(hostname, other_static_peers)) {
+          should_ignore = true;
+        }
+      }
+    }
+
+    return should_ignore;
+  }
+
+
+  bool is_static_peer(const std::string &hostname,
+                      const std::unordered_set<std::string> &other_static_peers) {
+
+    using namespace rmw_fastrtps_shared_cpp;
+    // Check if the host is a static peer on our list
+    auto aliases = utils::get_peer_aliases(hostname);
+    for (size_t ii = 0; ii < discovery_options_.static_peers_count; ++ii) {
+      if (hostname == discovery_options_.static_peers[ii].peer_address) {
+        RCUTILS_LOG_DEBUG_NAMED("rmw_fastrtps_shared_cpp",
+                                "Matching host in our static peer list");
+        return true;
+      }
+
+      if (aliases.count(discovery_options_.static_peers[ii].peer_address) > 0)
+      {
+        RCUTILS_LOG_DEBUG_NAMED("rmw_fastrtps_shared_cpp",
+                              "Matching host in our static peer list");
+        return true;
+      }
+    }
+
+    auto peer_aliases = utils::get_peer_aliases(my_hostname_);
+
+    for (const auto &alias: peer_aliases)
+    {
+      if (std::find(other_static_peers.begin(), other_static_peers.end(), alias) != other_static_peers.end()) {
+        RCUTILS_LOG_DEBUG_NAMED("rmw_fastrtps_shared_cpp",
+                                "Matched us in their static peer list");
+        return true;
       }
     }
+    return false;
   }
 
-  rmw_dds_common::Context * context;
-  const char * const identifier_;
+  rmw_dds_common::Context *context;
+  const char *const identifier_;
```

Comment:
Again this (odd) style changes make this much harder to review. It might actually be worth reworking this to not have any incidental style changes just so it's easier to understand later (as a separate commit).

Reply:
Removed whitespace changes in `custom_participant_info.hpp` in 65de1fb